spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DateFormatStringChecker.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * SpotBugs - Find bugs in Java programs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.StringAnnotation;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class DateFormatStringChecker extends OpcodeStackDetector {
+    private final BugReporter bugReporter;
+
+    /**
+     * Contains special flags that can trigger check (keywords); property of the rule
+     * (isRequired - whether flag should contain or be absent); flags to be checked
+     * and a bypassing flag (whether a rule can be skipped if such flag is included)
+     */
+    private static final class Rule {
+        boolean isRequired;
+        String ignore;
+        List<String> flags;
+        List<String> keywords;
+
+        Rule(List<String> flags, String ignore, boolean isRequired, List<String> keywords) {
+            this.flags = flags;
+            this.ignore = ignore;
+            this.isRequired = isRequired;
+            this.keywords = keywords;
+        }
+
+        boolean containsAny(String target, List<String> list) {
+            if (list == null) {
+                return false;
+            }
+            for (String item : list) {
+                if (target.contains(item)) {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
This may be a bit more elegant with streams like this:
```java
return list != null && list.stream().anyMatch(target::contains);
```
What do you think?

Reply:
Perfect, it is being used now